ccmbioinfo/lfs
# CARDS-2509: Reference question should support conditions based on souce form metadata

Add optional condition properties to reference questions:
- conditionalProperty: A property on the referenced form to be evaluated
- conditionalOperator: The operator that should be used for the comparison. Supports all operators supported by Conditional Sections
- conditionalValue: The value that the property should be compared against
- conditionalFallback: The value that should be saved instead of the referenced value if the condition is not satisfied
- conditionalType: The type of the property that is being compared

Refactored backend conditional section code in order to reuse conditional evaluation code

Update Reference Test forms to add testing support for new conditional references
- Added new section with 4 new questions to the bottom of ReferenceTestCopied. These questions are conditional on ReferenceTestUser being completed
- Made one question in ReferenceTestUser mandatory to allow the form to be edited between incomplete and complete


To test:
- Verify that form completion status works with mandatory questions in conditional sections as before as the code that updates completion status was adjusted in order to be reusable
- Verify that the new questions in ReferenceTestCopied show:
  - Fallback values when ReferenceTestUser is incomplete
  - The values from ReferenceTestUser when that form is complete
- Make sure to test reference answers on brand new forms and on forms that have been edited
